lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,296 @@
+	// subtracting the reserved amount to pay for a commitment transaction.
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+}
+
+// testChannelFundMax checks various channel funding scenarios where the user
+// instructed the wallet to use all remaining funds.
+func testChannelFundMax(net *lntest.NetworkHarness,
+	ht *harnessTest) {
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	carol := net.NewNode(
+		ht.t, "Carol", nil,
+	)
+	defer shutdownAndAssert(net, ht, carol)
+
+	dave := net.NewNode(
+		ht.t, "Dave", nil,
+	)
+	defer shutdownAndAssert(net, ht, dave)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	net.EnsureConnected(ht.t, carol, dave)
+
+	var testCases = []*chanFundMaxTestCase{
+		{
+			name:               "wallet amount is dust",
+			amount:             2_000,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount < min chan size (~18000sat)",
+			amount: 18_000,
+			// Using a feeRate of 1 sat/vByte ensures that we test
+			// for min chan size and not excessive fees.
+			feeRate:            1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount > min chan size (37000sat)",
+			amount: 37_000,
+			// The transaction fee to open the channel must be
+			// subtracted from Carol's balance.
+			// (since wallet balance < max-chan-size)
+			carolBalance: btcutil.Amount(37_000) -
+				fundingFee(1, false),
+		},
+
+		{
+			name: "wallet amount > max chan size " +
+				"(20000000sat)",
+			amount:       20_000_000,
+			carolBalance: lnd.MaxFundingAmount,
+		},
+
+		// Checks that to push corresponding amount to remote the local
+		// amount is not increased above MaxFundingAmount.
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount == max-chan-size",
+			amount:             20_000_000,
+			pushAmt:            lnd.MaxFundingAmount,
+			chanOpenShouldFail: true,
+		},
+
+		// Checks that enough local funds are kept in reserve before
+		// allowing to push amount to remote.
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount == max-chan-size - 1",
+			amount:             20_000_000,
+			pushAmt:            lnd.MaxFundingAmount - 1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount 16766000",
+			amount:       20_000_000,
+			pushAmt:      16_766_000,
+			carolBalance: lnd.MaxFundingAmount - 16_766_000,
+		},
+	}
+
+	for _, testCase := range testCases {
+		success := ht.t.Run(
+			testCase.name, func(tt *testing.T) {
+				runTestCase(net, ht, tt, carol, dave, testCase)
+			},
+		)
+
+		// Stop at the first failure. Mimic behavior of original test
+		// framework.
+		if !success {
+			break
+		}
+	}
+}
+
+// runTestCase runs a single test case asserting that test conditions are met.
+func runTestCase(net *lntest.NetworkHarness, ht *harnessTest,
```

Comment:
nit: all files in this directory share the same namespace. So maybe rename this to `runFundmaxTestCase()` to be a bit more specific?